### ISSUE TITLE
Getting typescript build error with the new discriminated union output

### ISSUE THREAD

**Issue by ronaldcurtis** [astahmer/openapi-zod-client#110]
Playground Link: [Here](https://openapi-zod-client.vercel.app/?doc=LQhQHsAcFMDsENIEsBcACAzAOgAxY6ErAGbgqhpoAuSVANtOgE7SThNUVoAm0AzgGMmSSDXCx0AFX6dKA8VXgCq5SpQQBbRmgBEOrpWgb4SOuj1c6SAXD6MDaTdoCqAOQAyASQDCAUVcAyr4AIlwAbtBMfEji6HgAjFjxoJDwVAAWfKq6APQsbBw5AN757FQA%2BkjcAL462ZSpVALp9WqKAOZZDmpowLpM4ACuVPwoKI3N5fIaWrBU%2Bj2UUJFpMbCe3OgT6eWlHFPgM3Cyi6lM8FojUa09fU7MrGWV3N2LRFtp6a89vILComt0AAlR4cNBVY5IYhISLfNQsACOgyQLE21CYg2gcMognSRngN0WlCoAE8YOg%2BFRhLB2ti1KQmMYVGhBsiXkTEZjKQAhcDcEmEuQKY6CtSISBWASrcQ5ABWfFidJxzXxoqJABIWMRzABiHLTNiwY58HK4-EmgAKaWaCyJaE5KOgaKpmNeLD4hrsXTtugATDgcHU6b8hCIxBI0Famuk0AaGCN2XadAAWHAANiDPsoIf%2B4fQrnAVDQShsoidr14kBYUoT6GI8DodledgEg2EpNFfUQSHKAGtoAK0ABtAC6oAN4mN2TNxm9agAgtxuAB5GDnPOvUnktDgABGsugygr-FDANiugAUgFl65I9aYwByeBLh87tfS2D21ju44f209B1UU7NBGi%2BH0%2BjCBtXSJKtlg4GE5xgz41WoMltEpalaSzHgT1zQFL2vVxgAtcAiCuNBILoTFhyBAAxbw0wAThweIRwACnSKgqEgLIchyKhwHARssBhKhiCwdh2hyTiNDoPJiAEJiWIASmodI0i-YhIjgGw%2BGLNA6HAGs1gAHVgAB3Wh0iINToGoeAmHaaAi24IzBlmItzLxFhbLfFZw3BPS1wZLRuCwf9FkozEUK3bQ9wPI9sJzMN8IANSguyBOLJdqHAXzFEc5yDKMj8sAcbwoBJVd-MBTc0PQeLDxOH5cJS88dCvG872jNAH3kSASVfOCPy-Kt%2BF-cMIsoQCnWA4gBg0OE%2BlA15YLXGhRjhebDhi%2Bq0AwogsKzZKzwjDrCOI0i5kiCiMpo%2BilNYjiuJ4lA%2BIEoS%2BBE5zxMk6SqFk%2BTFOY%2BJVIyDStW02BdP0wzjPEMysv6klbqouzto0cK4VA3btwOmlg1a07zE6oiSLIm6orsoc6IYkH2M47jeP4wThNE37HP%2BwGmAUx6wfUotIZYaH%2BFhkrwzMyyMhsjJMocpyXLcjy0C87TfOGgKkCCyIQqdLGegAWXACJqvXWqiVihr9ya48-jas7Se65peo0Y3oCG98AvyH85j-N1oCRR1NiWtAMZDlaYIGdaENFDHcfQqlDsJu3iYIm9LopphUeommHvp56mbelnPu%2BsSJM5mS5J54GVLUiHoC04WYfgYr4dgRG8tdiJs-Rhb9btHG6Ut-bE4JpKibzNOyau8iqfuumWIZl7mY%2BtmfvLqTK6Bvm68Fhuoeb1uP0lqyZbxezCsV1tldVnzZb8s3xECkCdfYUL%2B8oKNmkJYeHPOElXi0CMIhRYmoG66n1IcQ0xpTQqlnDkL%2B6RTZ%2Bx6Ig5BG4iSTmXNqEODY6DYJQpQPoYDtS6D1BOI0cwTQzngCaRcK5PZrCmnaPow9GqJWwvCAOyIgJKkWH0KAfCGhR0iBtEBnC3yELtMPfGR0JE9DgO5KR4FsqJiJF2Og%2BCcESOIVqCBFCYE0JNCCLu0B0FML4awva7DmqcJmsHeRagBGQCESBER8FNqOKWJAZRiwZGjzkV4xRGhfHqK-KY3BmiCGWLQCQ-RUDJxUNgXieBIIJRKDMYw8QzCwlsOthwiR9jQm9DfK4taoiY6uO8cU4ke1ZFVLUMEmpTjRp0AyZErRUjdHgLIZAjQ0CklGJyEbE2WTYA5P4ahbcNjXFFIac4sp7ixHFKgM0-xmEGmGFgEozZLSIkqLwdEnRsS9G9IMYMuBtCcgVQGuY7JMS8kJVsdhOZXiSmCMceUjx4jOGrN2espObytk7KBS05GHSjmcO6aQnQ5CEmUKoNQy5JppCUjueMh51j8nPKzK8rxCzPlLMqW8v5oKAVjyBU03ZRDqAyFtqeSejtEH7UOgwEC94H4fgrFrUMGgiBpHYKKL5pJXAXDii4noJi3bop-lip59K8LtSZRyh8LBTEexqk-b241faTX9oHXhKiI6nCJZ4geyEh51ICcnBl%2BFzrp3JtdLOc9c4LyeozV671WZfXZhvLmVdeYg35vXRuOlRYtzhsfCyp9Pz3wKgrHgStjgq28plc%2BGs1jP2Cm-PWEU0ltJsDKuq0zsUKvtiTQiTtHz5ALe7TlXtvw6sUHqjk3Cg7AWNWEqmq1TU-IaBa7C5LAlEhOpPe109M493no9JehcvUl19X9Le1cd7gz3qGkWekI3i1MtG6Wsbz7xqKq5a%2Bybb5prshmp%2BWsX48xzWFCZVMUJYO0VCqZcVS3HKHZi7c2yNC7lhF%2Bvau5WbQHgLAG1iqIzpTRrlVpGSVZWTKj0VFVAi0WzlTbEdE87XKp6g%2BEYlINWP0-NquwuqLGtoNbNcOnwQ7dsjnBZZ2MB1Zm-ePFOY7SYZydVO11M6C6euLmvMuS6AYBprqDXemkD7hqPhLPd1kD1y0vom09cwU1q3vlez8N7s2MlzXCJ9lqS3yo47a88MHqJZUI0WeA7QTCwEpGVFsbZaACi4F2ZAfYBzoFHEAA)

**Error output**:
```
error TS2740: Type 'ZodIntersection<ZodObject<{ path: ZodString; value: ZodObject<{}, "strip", ZodTypeAny, {}, {}>; }, "strip", ZodTypeAny, { path: string; value: {}; }, { path: string; value: {}; }>, ZodObject<...>>' is missing the following properties from type 'ZodObject<{ op: ZodTypeAny; } & ZodRawShape, UnknownKeysParam, ZodTypeAny, { [x: string]: any; op?: any; }, { [x: string]: any; op?: any; }>': _cached, _getCached, shape, strict, and 14 more.

```

Hi there, we seem to be getting an error with the latest release (v1.6.1) with the introduced `z.discriminatedUnion` output. We have an implementation of [JSONPatch](https://jsonpatch.com/) on one of our open-API endpoints, and the error seems to be centered around this.

Any help is appreciated! 🙏

**Comment by astahmer** [comment]
hey, @dominik-parkopedia any idea on what's needed to fix this ? mentioning you since you introduced z.discriminatedUnion [here](https://github.com/astahmer/openapi-zod-client/pull/106)

**Comment by dominik-parkopedia** [comment]
@astahmer @ronaldcurtis Very interesting … Sorry for breaking your build! I won't have time today, but will have a look tomorrow. Seems like we have to increase test coverage.

**Comment by ronaldcurtis** [comment]
@astahmer @dominik-parkopedia Looks like this is a known problem: https://github.com/colinhacks/zod/issues/2121

Also `disciminatedUnion` is planned to become deprecated: https://github.com/colinhacks/zod/discussions/2109

TLDR:
> z.discrimininatedUnion is too fragile and causes too much confusion so it needs to go.

Perhaps it might be worth reverting the change? And waiting for the Zod team to introduce the new `z.switch` api?

**Comment by astahmer** [comment]
that's fine with me 👍  thanks for the investigation btw

**Comment by dominik-parkopedia** [comment]
Sorry for only coming back today. I will have a look at that!
